utils/pattern/src/frontend/mod.rs

```diff
@@ -170,7 +176,7 @@ where
 #[cfg(feature = "alloc")]
 impl<B> Pattern<B, <B::Store as ToOwned>::Owned>
 where
-    B: PatternBackend,
+    B: PatternBackend<StoreUtf8Error = Utf8Error>,
```

Comment:
Issue: No, this is not right. You should add a generic type `E` instead of hard-coding that `StoreUtf8Error = Utf8Error` because it could be something else (like Infallible).

And put this in its own impl block that is not `feature = "alloc"` because this should work fine without `alloc`.

Reply:
> And put this in its own impl block that is not `feature = "alloc"` because this should work fine without `alloc`.

How could I add in it own block if it is part of `impl pattern`